mongo/src/test/java/org/axonframework/extensions/mongo/eventsourcing/eventstore/MongoOptionsFactoryTest.java

```diff
@@ -37,15 +39,17 @@ void setUp() {
 
     @Test
     void testCreateMongoOptions_defaults() {
-        MongoClientOptions options = factory.createMongoOptions();
-        MongoClientOptions defaults = MongoClientOptions.builder().build();
+        MongoClientSettings options = factory.createMongoOptions();
+        MongoClientSettings defaults = MongoClientSettings.builder().build();
 
-        assertEquals(defaults.getMaxWaitTime(), options.getMaxWaitTime());
-        assertEquals(defaults.getSocketTimeout(), options.getSocketTimeout());
-        assertEquals(defaults.getConnectionsPerHost(), options.getConnectionsPerHost());
-        assertEquals(defaults.getConnectTimeout(), options.getConnectTimeout());
-        assertEquals(defaults.getThreadsAllowedToBlockForConnectionMultiplier(),
-                     options.getThreadsAllowedToBlockForConnectionMultiplier());
+        assertEquals(defaults.getConnectionPoolSettings().getMaxWaitTime(TimeUnit.MILLISECONDS), options.getConnectionPoolSettings().getMaxWaitTime(TimeUnit.MILLISECONDS));
+        assertEquals(defaults.getSocketSettings().getConnectTimeout(TimeUnit.MILLISECONDS), options.getSocketSettings().getConnectTimeout(TimeUnit.MILLISECONDS));
+        assertEquals(defaults.getConnectionPoolSettings().getMaxSize(), options.getConnectionPoolSettings().getMaxSize());
+        // TODO connect time and socket timeout is identical
+//        assertEquals(defaults.getConnectTimeout(), options.getConnectTimeout());
+        // TODO deprecated
```

Comment:
We can drop this assertion as the `getThreadsAllowedToBlockForConnectionMultiplier` logic will be removed too.